src/App/Fossa/VSI/DynLinked/Internal/Resolve.hs

```diff
@@ -105,3 +117,69 @@ sortResolvedUnresolved = partitionEithers . map forkEither . toList
     forkEither dep = case dynamicDependencyResolved dep of
       Nothing -> Right $ dynamicDependencyDiskPath dep
       Just linuxPackage -> Left linuxPackage
+
+-- | Discover the linux distro under which we are currently executing.
+--
+-- Evaluates to @Nothing@ on non-Linux environments.
+-- Exits via @Diagnostics@ on parse errors.
+environmentDistro :: (Has Diagnostics sig m, Has ReadFS sig m) => m (Maybe LinuxDistro)
+environmentDistro
+  | not runningLinux = pure Nothing
+  | otherwise = recover $ readOsReleaseAt primaryPath <||> readOsReleaseAt fallbackPath
+  where
+    primaryPath :: Path Abs File
+    primaryPath = fsRoot </> $(mkRelFile "etc/os-release")
+
+    fallbackPath :: Path Abs File
+    fallbackPath = fsRoot </> $(mkRelFile "usr/lib/os-release")
+
+    readOsReleaseAt :: (Has Diagnostics sig m, Has ReadFS sig m) => Path Abs File -> m LinuxDistro
+    readOsReleaseAt file = do
+      content <- readContentsText file
+      readLinuxDistro content
+
+-- | Reads the linux distro from a file.
+readLinuxDistro :: (Has Diagnostics sig m) => Text -> m LinuxDistro
+readLinuxDistro contents = case runParser parseLinuxDistro "" filteredLines of
+  Left err -> fatal $ toText (errorBundlePretty err)
+  Right a -> pure a
+  where
+    prefixes :: [Text]
+    prefixes = map (<> "=") ["ID", "VERSION_ID"]
+
+    filteredLines :: Text
+    filteredLines = filterLinePrefixes contents
+
+    filterLinePrefixes :: Text -> Text
+    filterLinePrefixes = Text.unlines . filter lineIsUsed . fmap Text.strip . Text.lines
+      where
+        lineIsUsed line = any (`Text.isPrefixOf` line) prefixes
+
+type Parser = Parsec Void Text
+
+parseLinuxDistro :: Parser LinuxDistro
+parseLinuxDistro =
+  intercalateEffect sc $
```

Comment:
While I applaud the clever use of `toPermutation` here, another way to approach this might be to parse the file into a `Map Text Text`, and specifically `Map.lookup` the `ID` and `VERSION_ID` fields afterward